iib/workers/greenwave.py

```diff
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+from copy import deepcopy
+import json
+import logging
+
+import requests
+
+from iib.exceptions import IIBError
+from iib.workers.config import get_worker_config
+from iib.workers.tasks.utils import get_image_labels
+
+log = logging.getLogger(__name__)
+
+
+def gate_bundles(bundles, greenwave_config):
+    """
+    Check if all bundle images have passed gating tests in the CVP pipeline.
+
+    This function queries Greenwave to check if the policies are satisfied for each bundle image.
+
+    :param list bundles: a list of strings representing the pull specifications of the bundles to
+        be gated.
+    :param dict greenwave_config: the dict of config required to query Greenwave to gate bundles.
+    :raises IIBError: if any of the bundles fail the gating checks or IIB fails to get a
+        response from Greenwave.
+    """
+    conf = get_worker_config()
+    _validate_greenwave_params_and_config(conf, greenwave_config)
+
+    log.info('Gating on bundles: %s', ', '.join(bundles))
+    gating_unsatisfied_bundles = []
+    for bundle in bundles:
+        koji_build_nvr = _get_koji_build_nvr(bundle)
```

Comment:
We're making an assumption that the subject type is always koji-build here. I think this is fine for now as it's the only use case we have. But we do need some sort of mapping between a subject type and how to get the identifier from a bundle for it.

I think this is fine for this PR, but in the future we'll need a small change to address this.

Reply:
fwiw, I have added validation in `_validate_iib_api_config()` method to check if the subject-type is `koji_build`. If it is something else, IIB would not allow it for now since it doesn't support it.